### PR TITLE
fix Group & Ungroup with multi-selection

### PR DESCRIPTION
**Pull request by mrbuds**

### CODE REVIEW

File: WeakAurasOptions/WeakAurasOptions.lua [hunk 1]
```diff
@@ -4168,22 +4168,50 @@ function WeakAuras.EnsureDisplayButton(data)
...
     -- set dragging for selected buttons in reverse for ordering
-    for index=size,1,-1 do
-      local childId = tempGroup.controlledChildren[index]
-      local button = WeakAuras.GetDisplayButton(childId)
+    for index = size, 1, -1 do
+      local childId = tempGroup.controlledChildren[index];
+      local button = WeakAuras.GetDisplayButton(childId);
       button:SetDragging(data, drop, size);
-      children[childId] = true
+      children[childId] = true;
```
Comment (InfusOnWoW) [inline WeakAurasOptions/WeakAurasOptions.lua]: I can't express my love for this change in words. ❤️ ❤️ ❤️

**Comment by InfusOnWoW** [comment]
That's a bit incomplete, because of how it interacts with multi-selections that span top-level and grouped auras.

The ungroup/group actions are currently only shown for grouped/ungrouped auras and thus the code did not handled e.g. grouping a already grouped aura or trying to ungroup a ungrouped aura.

But with multi-selection, some of the selection might be in a different state from each other.

**Comment by Stanzilla** [comment]
Ace widget needs a minor version bump

**Comment by mrbuds** [comment]
Grouping / ungrouping a multi-selection of auras that are in a different state is working

Only thing i find a bit weird now is that if you select aura A (toplevel) + aura B (group G1) you can move both in a group by clicking the "group" action of aura A, but there is no group action for aura B

File: WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua [hunk 2]
```diff
@@ -514,18 +514,30 @@ local methods = {
...
-      self.callbacks.UpdateExpandButton();
-      self.grouping.parent = data.id;
+      if #self.grouping > 0 then
+        for index, childId in ipairs(self.grouping) do
+          tinsert(data.controlledChildren, childId);
+          local childButton = WeakAuras.GetDisplayButton(childId);
+          local childData = WeakAuras.GetData(childId);
+          childButton:SetGroup(data.id, data.regionType == "dynamicgroup");
+          childButton:SetGroupOrder(#data.controlledChildren, #data.controlledChildren);
+          childData.parent = data.id;
```
Comment (InfusOnWoW) [inline WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua]: This doesn't inform the old parent that the child is no longer part of its group. That breaks various things.

Comment (mrbuds) [reply WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua]: Indeed, i have submitted a fix

**Comment by emptyrivers** [comment]
What exactly does this fix?

**Comment by mrbuds** [comment]
Without this when you click on "group" or "ungroup" with more than one aura selected it works only on 1 aura

**Pull request merged**